eureka-core/src/main/java/com/netflix/eureka/registry/RemoteRegionRegistry.java

```diff
@@ -399,10 +401,10 @@ public boolean storeFullRegistry() {
      * @param delta - true, if the fetch needs to get deltas, false otherwise
      * @return - response which has information about the data.
      */
-    private Applications fetchRemoteRegistry(boolean delta) {
+    protected Applications fetchRemoteRegistry(boolean delta) {
         logger.info("Getting instance registry info from the eureka server : {} , delta : {}", this.remoteRegionURL, delta);
 
-        if (shouldUseExperimentalTransport()) {
+//        if (shouldUseExperimentalTransport()) {
```

Comment:
Functionally, this is probably the biggest change I've made, relying on the `EurekaHttpClient` interface rather that trying to upgrade the usage of the internal jersey1 apache HTTP client.

Reply:
It seems like the internal discovery client takes care of cross region identity header, but EurekaHttpClient doesn't?